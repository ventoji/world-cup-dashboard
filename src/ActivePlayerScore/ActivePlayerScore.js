import React, { useState } from 'react';
import { connect } from 'react-redux';
import InputWC from '../InputWC';

export const ActivePlayer = ({activeGames}) => {

  const [errors, setErrors] = useState('');
  const [foundPlayer, setFoundPlayer] = useState('');

  const handleActivePlayer = (event) => {
        const value = event.target.value.trim();
        const players = []

        const foundPlayer = activeGames.map((item) => {
            return [...players,item.homeTeam,item.awayTeam]
        })
        .flat()
        .filter(item => item === value);

        console.log(foundPlayer);
        setFoundPlayer(foundPlayer);
       // if(foundPlayer) setFoundPlayer(foundPlayer);
  }

    return (
        <>
            <InputWC
                htmlId="findTeam"
                label="fiend a team"
                type="text"
                placeholder="country name"
                name="homeTeam"
                onChange={handleActivePlayer}
                error={errors.hometeam}
            />
            {foundPlayer.length > 0 ? <span> {foundPlayer}</span>:<span>  </span>}
        </>
    );
};

const mapStatetoProps = (state) => {
    return {
      activeGames: state.activeGames,
    };
  };

  
  export default connect(mapStatetoProps)(ActivePlayer);
