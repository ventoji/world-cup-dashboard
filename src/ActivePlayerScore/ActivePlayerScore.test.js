import React from 'react';
// import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import { ActivePlayer } from './ActivePlayerScore';

describe('ActivePlayer component', () => {
  let component;
  let props;

  beforeEach(() => {
    props = {
      activeGames: ['Italy', 'Spain'],
    };

    component = shallow(<ActivePlayer {...props} />);
  });

  it('Should be defined', () => {
    expect(component).toBeDefined();
  });

  it('should render properly', () => {
    expect(toJson(component)).toMatchSnapshot();
    expect(component.find('InputWC')).toHaveLength(1);
  });

  //   it('should find a player that is actively playing', () => {

  //     //console.log('SHALLOW',toJson(component));
  //     console.log(toJson(component).children[0].props.onChange);
  //     component = shallow(<ActivePlayer {...props} />);
  //   //  console.log('MOUNT',toJson(component));
  //     const input = component.find('InputWC');

  //     input.simulate('change',{
  //         target: { value: 'Italy' }
  //     });
  //    // expect(toJson(component).children[0].props.onChange[0]).toHaveBeenCalled();
  //     console.log(input);
  //     expect(input.get(0).value).toEqual('Italy');

  //   });
});
