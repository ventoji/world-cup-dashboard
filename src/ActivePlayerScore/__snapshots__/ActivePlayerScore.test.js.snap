// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivePlayer component should render properly 1`] = `
<Fragment>
  <InputWC
    htmlId="findTeam"
    label="fiend a team"
    name="homeTeam"
    onChange={[Function]}
    placeholder="country name"
    required={true}
    type="text"
  />
  <span>
     
  </span>
</Fragment>
`;
